doc/source/development/rfc/rfc99_geometry_coordinate_precision.rst

```diff
@@ -0,0 +1,448 @@
+On a test dataset, setting a 1 mm precision reduced the size of the .zip of the
+.gpkg file from 766 MB to 667 MB (13% size decrease).
+If only a metric precision is wished, this increases to 26 useless least-significant bits.
+
+.. code-block:: python
+
+    >>> import math
+    >>> earth_radius_in_metre = 6378137
+    >>> one_degree_in_metre = earth_radius_in_metre * math.pi / 180.0
+    >>> mm_prec_in_degree = 1e-3 / one_degree_in_metre
+    >>> print(mm_prec_in_degree)
+    8.983152841195215e-09
+    >>> max_integer_part = 180  # for coordinates in range [-180,180]
+    >>> significant_bits_needed = math.ceil(math.log2(max_integer_part)) + math.ceil(math.log2(1 / mm_prec_in_degree)) + 1
+    >>> mantissa_bit_size = 52
+    >>> unused_bits = mantissa_bit_size - significant_bits_needed
+    >>> print(unused_bits)
+    16
+
+
+C and C++ API extensions and changes
+------------------------------------
+
+A new ``OGRGeomCoordinatePrecision`` class is introduced:
+
+.. code-block:: c++
+
+    /** Geometry coordinate precision.
+     *
+     * This may affect how many decimal digits (for text-based output) or bits
+     * (for binary encodings) are used to encode geometries.
+     *
+     * It is important to note that the coordinate precision has no direct
+     * relationship with the "physical" accuracy. It is generally advised that
+     * the resolution (precision) be at least 10 times smaller than the accuracy.
+     */
+    struct CPL_DLL OGRGeomCoordinatePrecision
+    {
+        static constexpr double UNKNOWN = 0;
+
+        /** Resolution for the coordinate precision of the X and Y coordinates.
+         * Expressed in the units of the X and Y axis of the SRS.
+         * For example for a projected SRS with X,Y axis unit in metre, a value
+         * of 1e-3 corresponds to a millimetric precision.
+         * For a geographic SRS (on Earth) with axis unit in degree, a value
+         * of 8.9e-9 (degree) also corresponds to a millimetric precision.
+         * Set to 0 if unknown.
+         */
+        double dfXYResolution = UNKNOWN;
+
+        /** Resolution for the coordinate precision of the Z coordinate.
+         * Expressed in the units of the Z axis of the SRS.
+         * Set to 0 if unknown.
+         */
+        double dfZResolution = UNKNOWN;
+
+        /** Resolution for the coordinate precision of the M coordinate.
+         * Set to 0 if unknown.
+         */
+        double dfMResolution = UNKNOWN;
+
+        /** Map from a format name to a list of format specific options.
+         *
+         * This can be for example used to store FileGeodatabase
+         * xytolerance, xorigin, yorigin, etc. coordinate precision grids
+         * options, which can be help to maximize preservation of coordinates in
+         * FileGDB -> FileGDB conversion processes.
+        std::map<std::string, CPLStringList> oFormatSpecificOptions{};
+
+        /** Helper to specify X,Y,Z precision in terms of metric resolution.
```

Comment:
Is this correct?

```suggestion
        /** Helper to specify X,Y,Z precision in meters
```

Reply:
I would consider using this globally. I'm not familiar with the term "metric resolution."

Reply:
done